File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        vector<bool>seen(nums.size() , false); 

        for( int i = 0; i< nums.size() ; i++){
            if(seen[nums[i]] == true) return nums[i];

            seen[nums[i]] = true; 
        }
        return -1; 
        
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        HashSet<Integer>st = new HashSet<>(); 
        for( int num : nums){
            if(st.contains(num)) return num; 
            else  st.add(num);
        }
        return 0;
        
    }
}